plugins/module_utils/foreman_helper.py

```diff
@@ -532,24 +532,24 @@ def _delete_entity(self, resource, current_entity, params, synchronous):
         payload = {'id': current_entity['id']}
         if params:
             payload.update(params)
-        entity = self.resource_action(resource, 'destroy', payload, synchronous=synchronous)
+        entity = self.resource_action(resource, 'destroy', payload)
 
         # this is a workaround for https://projects.theforeman.org/issues/26937
         if entity and 'error' in entity and 'message' in entity['error']:
             self.fail_json(msg=entity['error']['message'])
 
         return None
 
-    def resource_action(self, resource, action, params, options=None, data=None, files=None, synchronous=True, ignore_check_mode=False, record_change=True):
+    def resource_action(self, resource, action, params, options=None, data=None, files=None, ignore_check_mode=False, record_change=True):
         resource_payload = self._resource_prepare_params(resource, action, params)
         if options is None:
             options = {}
         try:
             result = None
             if ignore_check_mode or not self.check_mode:
                 result = self._resource_call(resource, action, resource_payload, options=options, data=data, files=files)
-                is_foreman_task = isinstance(result, dict) and 'action' in result and 'state' in result and 'pending' in result
-                if synchronous and is_foreman_task:
+                is_foreman_task = isinstance(result, dict) and 'action' in result and 'state' in result and 'started_at' in result
```

Comment:
why the change from pending to started at?

Reply:
Oh, so you spotted it...
`pending` seems to be added by a special katello renderer `started_at` is available on all tasks i could find.

Reply:
WFM. Was just wondering :)